### ISSUE TITLE
I don't get it

### ISSUE THREAD

**Issue by TrollWarlord** [tleyden/couchbase-cluster-go#1]
The code is kinda hard to follow. It seems like there's a "warmup" routine and a "start" routine. During the start routine, why can't I specify etcdservers ie: 

[Unit]
Description=cb

[Service]
TimeoutStartSec=0
EnvironmentFile=/etc/environment
ExecStartPre=-/usr/bin/docker kill couchbase
ExecStartPre=-/usr/bin/docker rm couchbase
ExecStartPre=/usr/bin/docker pull tleyden5iwx/couchbase-server-3.0.1
ExecStart=/usr/bin/docker run --name couchbase -v /opt/couchbase/var:/opt/couchbase/var --net=host tleyden5iwx/couchbase-server-3.0.1 couchbase-cluster start-couchbase-node --local-ip=172.16.2.102 --etcd-servers=172.16.2.102:4001
ExecStop=/usr/bin/docker stop couchbase
Restart=on-failure
RestartSec=5

**Comment by TrollWarlord** [comment]
Can we pluse use os.Flag too!

**Comment by tleyden** [comment]
> why can't I specify etcdservers ie

--etcd-servers=172.16.2.102:4001 is an optional param:

https://github.com/tleyden/couchbase-cluster-go/blob/master/cmd/couchbase-cluster/main.go#L15

actually I'm not really sure what you're asking.

**Comment by tleyden** [comment]
> It seems like there's a "warmup" routine and a "start" routine.

Do you mean `ExecStartPre` vs `ExecStart`?

**Comment by tleyden** [comment]
> Can we pluse use Flag package too!

I'm using [docopt.go](https://github.com/docopt/docopt.go), and so far it's working out.

**Comment by tleyden** [comment]
I'm gonna close the ticket for now since I don't really see anything actionable, but feel free to continue the discussion here ..

**Comment by TrollWarlord** [comment]
The question is in "start-couchbase-node" the params start-couchbase-node --local-ip=... works but if you add --etcd-servers=... it fails to parse the etcd-servers.

**Comment by TrollWarlord** [comment]
this package isn't populating in etcd the keys for /couchbase.com/couchbase-node-state which is causing timeouts. I'm trying to resolve this since the package sees 0 etcd nodes. I'm trying to pass in the servers via params but it doesn't parse properly.

**Comment by tleyden** [comment]
> The question is in "start-couchbase-node" the params start-couchbase-node --local-ip=... works but if you add --etcd-servers=... it fails to parse the etcd-servers.

OK that makes sense.  I will push a fix.

**Comment by tleyden** [comment]
@TrollWarlord I pushed a fix, but I haven't retested yet.  Let me know if it fixes your issue.

**Comment by TrollWarlord** [comment]
Even though the documentation is wrong, the issue is that --etcd-servers is NOT properly parsed.
/usr/bin/docker run --name couchbase -v /opt/couchbase/var:/opt/couchbase/var --net=host tleyden5iwx/couchbase-server-3.0.1 couchbase-cluster start-couchbase-node --local-ip=172.16.2.102 --etcd-servers=172.16.2.102:4001

Is the example (provided above) of me trying to call --etcd-servers with it which causes the application to just print out the usage example

**Comment by tleyden** [comment]
That's more than just documentation, its a code generation spec.  

I just tested and it worked for me:

```
# go run main.go start-couchbase-node --local-ip 127.0.0.1 --etcd-servers "128.0.0.1"
2015/02/07 03:00:36 Connect to explict etcd servers: [128.0.0.1]
etc..
```

**Comment by TrollWarlord** [comment]
Ah gotcha, my mistake. I id see that after the fact (looked through docopt) that's kinda weird but I guess it works! I'll test it now!

**Comment by tleyden** [comment]
Cool!  Yep, docopt has a much different approach ..

**Comment by TrollWarlord** [comment]
GoDep isn't updated so it's not pulling the new version in prod!

**Comment by tleyden** [comment]
How are you running this?  via the docker image or directly on your system?

**Comment by TrollWarlord** [comment]
Via a fleet system file that pulls and runs the docker image. But it's not pulling your latest version. I assume it's godeps pulilng the tag you set in there and not the new one.

**Comment by tleyden** [comment]
I think the docker image just needs to be rebuilt.  Btw which instructions are you following, and are you interested in couchbase server + sync gateway, or just couchbase server?

**Comment by tleyden** [comment]
I just rebuilt new [docker images](https://registry.hub.docker.com/u/tleyden5iwx/couchbase-cluster-go/builds_history/135098/), can you try again?

**Comment by tleyden** [comment]
If you only need couchbase server, you probably want to follow the instructions in this [blog post](http://tleyden.github.io/blog/2014/11/01/running-couchbase-cluster-under-coreos-on-aws/) (recently updated a few minutes ago)

The main command to kick things off is now:


```
$ sudo docker run --net=host tleyden5iwx/couchbase-cluster-go:0.4 couchbase-fleet launch-cbs --version 3.0.1 --num-nodes 3 --userpass "user:passw0rd" --docker-tag 0.4
```

Where:

* **--version** the version of Couchbase Server to use.  Valid values are 3.0.1 or 2.2.0.  (community edition in both cases)
* **--num-nodes** the total number of couchbase nodes to start -- should correspond to number of ec2 instances (eg, 3)
* **--userpass** the username and password as a single string, delimited by a colon (:) 
* **--docker-tag** the docker tag to use to launch containers
